### PR TITLE
UPSTREAM: <drop>: Bump golang.org/x/net to v0.23.0

### PR DESCRIPTION
**Pull request by flavianmissi**
./hack/pin-dependency.sh golang.org/x/net v0.23.0
./hack/update-vendor.sh

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -581,7 +581,7 @@ replace (
 	golang.org/x/lint => golang.org/x/lint v0.0.0-20190930215403-16217165b5de
 	golang.org/x/mobile => golang.org/x/mobile v0.0.0-20201217150744-e6ae53a27f4f
 	golang.org/x/mod => golang.org/x/mod v0.8.0
-	golang.org/x/net => github.com/openshift-priv/golang-net v0.0.0-20240328080036-60d0f00ca866
```
Comment (flavianmissi) [inline go.mod:584]: I manually removed this, but seems like one of the vendoring scripts added it back but replaced golang.org/x/net with golang.org/x/net... do we care?

**Review by soltysh — APPROVED** [review]
/lgtm 
/approve

/remove-label backports/unvalidated-commits
/label backports/validated-commits

**Comment by soltysh** [comment]
/label backport-risk-assessed

Manually overriding the valid-bug jira, since that is just updating to use the officially released version.
/label jira/valid-bug

**Pull request merged**